src/basic/selinux-util.c

```diff
@@ -103,37 +108,95 @@ int mac_selinux_init(void) {
                   format_timespan(timespan, sizeof(timespan), after_timestamp - before_timestamp, 0),
                   (l+1023)/1024);
 
+        return 0;
+}
+#endif
+
+int mac_selinux_init(void) {
+#if HAVE_SELINUX
+        int r;
+
+        if (initialized)
+                return 0;
+
+        if (!mac_selinux_use())
+                return 0;
+
+        r = selinux_status_open(/* no netlink fallback */ 0);
+        if (r < 0) {
+                return log_enforcing_errno(errno, "Failed to open SELinux status page: %m");
+        }
+
+        last_policyload = selinux_status_policyload();
+
+        r = open_label_db(&label_hnd);
+        if (r < 0) {
+                selinux_status_close();
+                return r;
+        }
+
+        enforcing_status_func = selinux_status_getenforce;
```

Comment:
a comment explaining here what this is all about would be great

Reply:
added comment